src/playlist.js

```diff
@@ -305,6 +307,43 @@ const calculateExpiredTime = function(playlist, expiredSync, segmentSync) {
 
     return segmentSync.time - sumDurations(playlist, syncIndex, 0);
   }
+  return 0;
```

Comment:
That is a good point, but I don't think returning 0 is the right call here since there are live scenarios where we may not have any sync points, but there has been expired time, we just can't calculate it. Since `endList` indicates a VOD asset, we actually already have a known sync point then just off knowing `endList` alone, which would be `{ time: 0, mediaSequence: 0 }` since VOD always starts at time 0 segment 0. I'd say it would make most sense to update `getPlaylistSyncPoints` to set `expiredSync` to `{ time: 0, mediaSequence: 0 }` if  `playlist.endList`, otherwise set to `playlist.syncInfo`, and finally resorting to `null`. That way `calculateExpiredTime` can properly calculate for VOD case, while still returning `null` in the live case where we don't have enough information